ethcore/src/miner/miner.rs

```diff
@@ -614,12 +615,19 @@ impl Miner {
 							}
 						}).unwrap_or(default_origin);
 
+						// try to install zgp-checker before appending transactions
```

Comment:
There are 2 concerns about this:
1) set_zero_gas_price_checker() must be called on Miner() initialization + on new blocks, but when Miner is initialized, Client (which is used by `OnChainZeroGasPriceChecker::from_chain`) is not yet ready => I placed it here
2) as I understand, registry can be updated => address can be changed => possibly this condition `if !transaction_queue.is_zero_gas_price_checker_set()` is unnecessary. But I have checked how updater works:
https://github.com/ethcore/parity/blob/master/updater/src/updater.rs#L242
it works the same way. So maybe I'm wrong && it is ok to have single initialization.